packages/launcher/src/renderer/launcher/wallets/WalletsView.js

```diff
@@ -84,10 +94,64 @@ const addWalletMutation = graphql`
   }
 `
 
+const TOKEN_ADDRESS = {
+  testnet: '0xa46f1563984209fe47f8236f8b01a03f03f957e4',
+  mainnet: '0xdf2c7238198ad8b389666574f2d8bc411a4b7428',
+}
+
+const newEthRequest = (id, method, params) => {
+  return {
+    id: id,
+    jsonrpc: '2.0',
+    method,
+    params,
+  }
+}
+
 class WalletsView extends Component<Props, State> {
   static contextType = EnvironmentContext
 
-  state = {}
+  state = {
+    balances: {},
+  }
+
+  componentDidMount() {
+    this.getBalances()
+  }
+
+  async getBalances() {
+    const { ethWallets } = this.props.wallets
+    const balances = {}
+    for (let i = 0; i < ethWallets.hd[0].accounts.length; i++) {
+      const account = ethWallets.hd[0].accounts[0]
+      try {
+        // TODO: determine eth network
+        const ethBalanceReq = newEthRequest(1, 'eth_getBalance', [
```

Comment:
The `id` argument here should be a random string or an incremental number rather than a hardcoded one, it could create weird errors otherwise.